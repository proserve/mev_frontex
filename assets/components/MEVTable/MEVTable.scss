.trans-list {
  margin-top: 20px;
  overflow: auto;
    overflow-y: hidden;
  .header {
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
    h2 {
      color: #f7c057;
      text-transform: uppercase;
      font-weight: 400;
    }

    .search-form {
      position: relative;
      display: flex;
      align-items: center;

      .SVGInline {
        position: absolute;
        left: 20px;
        padding-top: 5px;
      }

      input {
        border-radius: 20px;
        line-height: 20px;
        padding: 10px 10px 10px 40px;
        font-size: 16px;
        border: 1px solid #eeeeee;
        width: 250px;
        transition: width 0.5s;
        &:focus {
          outline: none;
          width: 300px;
        }
      }
    }
  }
  table {
    width: 100%;
    text-align: left;
    border-collapse: collapse;
    line-height: 20px;
    .spacer {
      height: 10px;
      &:first-child {
        height: 20px;
      }
    }

    tr.head {
      padding-bottom: 10px;
      margin-bottom: 10px;

      th {
        cursor: pointer;
        padding-bottom: 10px;
        border-bottom: 1px #cccccc solid;
        .SVGInline{
          margin-left: 10px;
        }
      }
    }

    tbody tr {
      margin: 10px;
      padding: 10px;

      td {
        padding: 10px;
        background-color: white;
        .red-circle{
          display: inline-flex;
          width: 12px;
          height: 12px;
        }
      }
    }
  }
}
