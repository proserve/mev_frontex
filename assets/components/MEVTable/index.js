import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import SVG from 'react-svg-inline';

import searchIcon from '../../static/icons/search.svg';
import sortDown from '../../static/icons/sort-down.svg';
import sortUp from '../../static/icons/sort-up.svg';
import './MEVTable.scss';
import { HEADERS, sort, search } from '../../common/utils';


export default class MEVTable extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      transactions: props.transactions,
      filter: '',
      sortKey: '',
      direction: '',
    };
    this.onChangeSearch = this.onChangeSearch.bind(this);
  }

  onChangeSearch(e) {
    const { transactions } = this.props;
    const filter = e.target.value;
    this.setState({ transactions: search(filter, transactions), filter });
  }

  onSort(sortKey) {
    const { transactions, direction } = this.state;
    const newDirection = direction === 'ASC' ? 'DESC' : 'ASC';
    const newTrans = sort(transactions, newDirection, sortKey);
    this.setState({ transactions: newTrans, direction: newDirection, sortKey });
  }


  render() {
    const {
      transactions, filter, sortKey, direction,
    } = this.state;
    return (
      <div className="trans-list">
        <div className="header">
          <h2>Transactions</h2>
          <div className="search-form">
            <SVG svg={searchIcon} />
            <input type="search" placeholder="Search" value={filter} onChange={this.onChangeSearch} />
          </div>
        </div>
        <table>
          <thead>
            <tr className="head">
              {HEADERS.map(({ value, label }) => (
                <th key={value} onClick={() => this.onSort(value)}>
                  {label}
                  {sortKey === value ? <SVG svg={direction === 'ASC' ? sortUp : sortDown} /> : null}
                </th>
              ))}
            </tr>
          </thead>
          <tbody>
            {transactions.map(transaction => (
              <Fragment key={transaction.referenceNo}>
                <tr className="spacer" />
                <tr>
                  {HEADERS.map(({ value }) => (
                    <td key={transaction[value]}>
                      {value === 'date' && <div className="red-circle" />}
                      <span>
                        {(value === 'totalAmount' || value === 'balance') && `${transaction.currency} `}
                      </span>
                      {transaction[value]}
                    </td>))}
                </tr>
              </Fragment>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}

MEVTable.propTypes = {
  transactions: PropTypes.arrayOf(PropTypes.object),
};

MEVTable.defaultProps = {
  transactions: [],
};
