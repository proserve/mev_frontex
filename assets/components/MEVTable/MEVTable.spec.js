import React from 'react';
import 'isomorphic-fetch';
import { shallow } from 'enzyme';
import MEVTable from './index';

describe('App', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<MEVTable />);
    expect(component).toMatchSnapshot();
  });
});
