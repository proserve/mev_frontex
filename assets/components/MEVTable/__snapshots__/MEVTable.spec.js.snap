// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly in "debug" mode 1`] = `
<div
  className="trans-list"
>
  <div
    className="header"
  >
    <h2>
      Transactions
    </h2>
    <div
      className="search-form"
    >
      <SVGInline
        classSuffix="-svg"
        cleanup={Array []}
        cleanupExceptions={Array []}
        component="span"
        svg={null}
      />
      <input
        onChange={[Function]}
        placeholder="Search"
        type="search"
        value=""
      />
    </div>
  </div>
  <table>
    <thead>
      <tr
        className="head"
      >
        <th
          key="date"
          onClick={[Function]}
        >
          Date
        </th>
        <th
          key="referenceNo"
          onClick={[Function]}
        >
          Reference No
        </th>
        <th
          key="orderNo"
          onClick={[Function]}
        >
          OrderNo
        </th>
        <th
          key="paymentType"
          onClick={[Function]}
        >
          Payment Type
        </th>
        <th
          key="totalAmount"
          onClick={[Function]}
        >
          Total Amount
        </th>
        <th
          key="balance"
          onClick={[Function]}
        >
          Balance
        </th>
        <th
          key="status"
          onClick={[Function]}
        >
          Status
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
